File: Java/linked-list-cycle.java
/**
 * Problem Link: https://leetcode.com/problems/linked-list-cycle/
 * 
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode node1 = head;
        ListNode node2 = null;
        if(head != null && head.next != null) {
            node2 = head.next;
        } else {
            return false;
        }
        while(node1 != null) {
            if(node1 == node2) return true;
            node1 = node1.next;
            if(node2.next != null && node2.next.next != null) node2 = node2.next.next;
            else return false;
        }
        return false;
    }
}

File: Swift/linked-list-cycle.swift
/**
 * https://leetcode.com/problems/linked-list-cycle/
 * 
 * 
 */ 
// Date: Tue May  5 13:30:25 PDT 2020
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */

class Solution {
    func hasCycle(_ head: ListNode?) -> Bool {
        var fast = head?.next
        var slow = head
        while fast != nil, fast?.next != nil {
            if fast === slow { return true }
            slow = slow?.next
            fast = fast?.next?.next
        }
        return false
    }
}
